test/Test.Amqp.NetMF/LinkTests.cs

```diff
@@ -650,5 +650,32 @@ public void TestMethod_AdvancedLinkFlowControl()
             session.Close();
             connection.Close();
         }
+
+        /// <summary>
+        /// This test proves that issue #14 is fixed.
+        /// https://github.com/Azure/amqpnetlite/issues/14
+        /// </summary>
+ #if !(NETMF || COMPACT_FRAMEWORK)
+        [TestMethod, ExpectedException(typeof(ArgumentException))]
```

Comment:
LinkTests.cs is also used by the NETMF test projects. Since NETMF does not support MSTest attributes, the method name is used to indicate a test case (name starts with "TestMethod_"). The Test.Amqp.NetMF42/43 projects are NETMF applications and they use reflection to find the test cases to run.

For this reason we cannot use ExpectedException attribute or the test case will fail on NETMF. Catch the exception explicitly and verify.